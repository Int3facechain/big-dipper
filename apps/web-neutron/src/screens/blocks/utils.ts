import { ItemType } from '@/screens/blocks/types';

export const resolveProfile = (item: ItemType, profile: AvatarName) => ({
  ...profile,
  name: item.consumerOperatorAddress,
  address: item.consumerOperatorAddress,
});
