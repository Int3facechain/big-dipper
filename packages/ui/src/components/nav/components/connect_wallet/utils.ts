export const walletConnectRelayURL =
  process?.env?.NEXT_PUBLIC_WALLET_CONNECT_RELAY_URL ?? 'wss://relay.walletconnect.com';
export const walletConnectProjectID =
  process?.env?.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID ?? '479ef4c34890ad2226193483069863c4';
export const bigDipperRepositoryURL = 'https://github.com/forbole/big-dipper-2.0-cosmos';
export const bigDipperIcon =
  'https://github.com/forbole/big-dipper-2.0-cosmos/blob/main/apps/web/public/icons/favicon-32x32.png?raw=true';
