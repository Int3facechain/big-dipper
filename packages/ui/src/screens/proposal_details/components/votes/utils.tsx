import type { VoteType } from '@/screens/proposal_details/components/votes/types';

export const getVoteKey = (vote: string) => {
  const votes = {
    yes: 'yes',
    no: 'no',
    VOTE_OPTION_NO_WITH_VETO: 'veto',
    abstain: 'abstain',
    NOT_VOTED: 'notVoted',
  };

  return votes[vote as keyof typeof votes] || vote;
};

export const filterDataByTab = (props: { data: VoteType[]; notVoted: VoteType[]; tab: number }) => {
  if (props.tab === 5) {
    return props.notVoted;
  }

  return props.data.filter((x) => {
    if (props.tab === 1) {
      return x.vote === 'yes';
    }

    if (props.tab === 2) {
      return x.vote === 'no';
    }

    if (props.tab === 3) {
      return x.vote === 'abstain';
    }

    return true;
  });
};
