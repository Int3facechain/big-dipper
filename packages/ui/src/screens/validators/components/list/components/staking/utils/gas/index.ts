import { SigningStargateClient, calculateFee, GasPrice } from '@cosmjs/stargate';

export const estimateFee = async (
  client: SigningStargateClient,
  address: string,
  msgs: any[],
  memo: string,
  denom: string
) => {
  const gasPrice = GasPrice.fromString(`0.01${denom}`);
  const gasEstimation = await client.simulate(address, msgs, memo);
  const fee = calculateFee(Math.round(gasEstimation * 1.3), gasPrice);
  return fee;
};
