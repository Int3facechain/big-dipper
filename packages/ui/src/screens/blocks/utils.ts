import { ItemType } from '@/screens/blocks/types';

export const resolveProfileAddress = (_item: ItemType, profile: AvatarName) => profile;
