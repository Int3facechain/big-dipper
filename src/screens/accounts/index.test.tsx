import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import List from './components/list';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
}));

jest.mock('./components/list', () => (props) => <div id="List" {...props} />);

// ==================================
// unit tests
// ==================================
describe('screen: Accounts', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <MockTheme>
        <List />
      </MockTheme>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
