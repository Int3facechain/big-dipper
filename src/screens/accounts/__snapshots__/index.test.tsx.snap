// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Accounts matches snapshot 1`] = `
<div
  id="List"
/>
`;
